File: svws-asd/src/main/java/de/svws_nrw/asd/validate/lehrer/ValidatorLsv01LehrerStammdatenVorname.java
package de.svws_nrw.asd.validate.lehrer;

import java.util.function.Supplier;

import de.svws_nrw.asd.validate.Validator;
import de.svws_nrw.asd.validate.ValidatorKontext;
import jakarta.validation.constraints.NotNull;

/**
 * Dieser Validator führt eine Statistikprüfung auf den Vornamen bei den Stammdaten
 * eines Lehrers einer Schule aus.
 */
public final class ValidatorLsv01LehrerStammdatenVorname extends Validator {

	/** Der Lehrer-Vorname */
	private final @NotNull Supplier<String> daten;

	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
	 *
	 * @param daten     der Vorname des Lehrers
	 * @param kontext   der Kontext des Validators
	 */
	public ValidatorLsv01LehrerStammdatenVorname(final @NotNull Supplier<String> daten,
			final @NotNull ValidatorKontext kontext) {
		super(kontext);
		this.daten = daten;
		_validatoren.add(new ValidatorLsv02LehrerStammdatenVorname(daten, kontext));
		_validatoren.add(new ValidatorLsv03LehrerStammdatenVorname(daten, kontext));
		_validatoren.add(new ValidatorLsv04LehrerStammdatenVorname(daten, kontext));
		_validatoren.add(new ValidatorLsv05LehrerStammdatenVorname(daten, kontext));
		_validatoren.add(new ValidatorLsv06LehrerStammdatenVorname(daten, kontext));
		_validatoren.add(new ValidatorLsv07LehrerStammdatenVorname(daten, kontext));
		_validatoren.add(new ValidatorLsv08LehrerStammdatenVorname(daten, kontext));

	}


	@Override
	protected boolean pruefe() {
		if (daten.get().trim().isBlank()) {
			addFehler(1, "Vorname der Lehrkraft: Der Vorname darf nicht nur aus Leerzeichen bestehen.");
			return false;
		}

		return true;
	}

}

File: svws-webclient/core/src/asd/validate/lehrer/ValidatorLsv01LehrerStammdatenVorname.ts
import { ValidatorLsv05LehrerStammdatenVorname } from '../../../asd/validate/lehrer/ValidatorLsv05LehrerStammdatenVorname';
import { ValidatorLsv08LehrerStammdatenVorname } from '../../../asd/validate/lehrer/ValidatorLsv08LehrerStammdatenVorname';
import { ValidatorLsv04LehrerStammdatenVorname } from '../../../asd/validate/lehrer/ValidatorLsv04LehrerStammdatenVorname';
import { ValidatorLsv02LehrerStammdatenVorname } from '../../../asd/validate/lehrer/ValidatorLsv02LehrerStammdatenVorname';
import { ValidatorLsv03LehrerStammdatenVorname } from '../../../asd/validate/lehrer/ValidatorLsv03LehrerStammdatenVorname';
import { LehrerStammdaten } from '../../../asd/data/lehrer/LehrerStammdaten';
import { ValidatorLsv06LehrerStammdatenVorname } from '../../../asd/validate/lehrer/ValidatorLsv06LehrerStammdatenVorname';
import { ValidatorLsv07LehrerStammdatenVorname } from '../../../asd/validate/lehrer/ValidatorLsv07LehrerStammdatenVorname';
import { Class } from '../../../java/lang/Class';
import { JavaString } from '../../../java/lang/JavaString';
import { ValidatorKontext } from '../../../asd/validate/ValidatorKontext';
import { Validator } from '../../../asd/validate/Validator';

export class ValidatorLsv01LehrerStammdatenVorname extends Validator {

	/**
	 * Die Lehrer-Stammdaten
	 */
	private readonly daten: LehrerStammdaten;


	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
	 *
	 * @param daten     die Daten des Validators
	 * @param kontext   der Kontext des Validators
	 */
	public constructor(daten: LehrerStammdaten, kontext: ValidatorKontext) {
		super(kontext);
		this.daten = daten;
		this._validatoren.add(new ValidatorLsv02LehrerStammdatenVorname(daten, kontext));
		this._validatoren.add(new ValidatorLsv03LehrerStammdatenVorname(daten, kontext));
		this._validatoren.add(new ValidatorLsv04LehrerStammdatenVorname(daten, kontext));
		this._validatoren.add(new ValidatorLsv05LehrerStammdatenVorname(daten, kontext));
		this._validatoren.add(new ValidatorLsv06LehrerStammdatenVorname(daten, kontext));
		this._validatoren.add(new ValidatorLsv07LehrerStammdatenVorname(daten, kontext));
		this._validatoren.add(new ValidatorLsv08LehrerStammdatenVorname(daten, kontext));
	}

	protected pruefe(): boolean {
		if (JavaString.isBlank(this.daten.vorname.trim())) {
			this.addFehler(1, "Vorname der Lehrkraft: Der Vorname darf nicht nur aus Leerzeichen bestehen.");
			return false;
		}
		return true;
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.asd.validate.lehrer.ValidatorLsv01LehrerStammdatenVorname';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.asd.validate.BasicValidator', 'de.svws_nrw.asd.validate.lehrer.ValidatorLsv01LehrerStammdatenVorname', 'de.svws_nrw.asd.validate.Validator'].includes(name);
	}

	public static readonly class = new Class<ValidatorLsv01LehrerStammdatenVorname>('de.svws_nrw.asd.validate.lehrer.ValidatorLsv01LehrerStammdatenVorname');

}

export function cast_de_svws_nrw_asd_validate_lehrer_ValidatorLsv01LehrerStammdatenVorname(obj: unknown): ValidatorLsv01LehrerStammdatenVorname {
	return obj as ValidatorLsv01LehrerStammdatenVorname;
}
